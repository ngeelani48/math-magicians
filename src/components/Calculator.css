.layout {
  display: flex;
  gap: 50px;
  padding: 0 50px;
  align-items: center;
}

.container {
  max-width: 40rem;
  min-width: 35rem;
  margin: auto;
  position: fixed;
  bottom: 15%;
  right: 2%;
  border: 2px solid #858694;
  padding: 1%;
}

.calc-title {
  text-align: center;
  font-size: 2rem;
  margin: 8%;
}

.result-display {
  background-color: #858694;
  color: #fff;
  font-size: 1.5rem;
  text-align: right;
  padding: 10px;
  height: 55px;
}

.calc-button-zero {
  grid-column: 1 / 3;
  font-size: 1rem;
  border: none;
  height: 70px;
}

.calc-button-right {
  background-color: #f5913e;
  font-size: 1.2rem;
}

.calc-button-right:hover {
  background-color: #ca603f;
}

.calc-buttons {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-gap: 2px;
}

.calc-button {
  font-size: 1rem;
  border: none;
  height: 70px;
}

.calc-button:hover,
.calc-button-zero:hover {
  background-color: #fff;
}

button {
  cursor: pointer;
}
