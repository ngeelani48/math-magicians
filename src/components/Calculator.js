import React from 'react';
import CalcDisplay from './CalcDisplay';

const Calculator = () => (
  <CalcDisplay />
);

export default Calculator;
