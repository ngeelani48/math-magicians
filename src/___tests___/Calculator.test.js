import React from "react";
import renderer from 'react-test-renderer';
import Calculator from "../components/Calculator";

it ('check if the calculator renders properly', ()=>{
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
});
