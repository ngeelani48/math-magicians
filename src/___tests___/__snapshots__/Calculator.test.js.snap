// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check if the calculator renders properly 1`] = `
<div
  className="layout"
>
  <h2
    className="calc-title"
  >
    Let's do some Math!
  </h2>
  <div
    className="container"
  >
    <div
      className="result-display"
    >
      0
    </div>
    <div
      className="calc-buttons"
    >
      <button
        className="calc-button"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="calc-button"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="calc-button"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="calc-button-right"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="calc-button"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="calc-button"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="calc-button"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="calc-button-right"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="calc-button"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="calc-button"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="calc-button"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="calc-button-right"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="calc-button"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="calc-button"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="calc-button"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="calc-button-right"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="calc-button-zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="calc-button"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="calc-button-right"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
